File: 409-longest-palindrome/longest-palindrome.js
/**
 * @param {string} s
 * @return {number}
 */
var longestPalindrome = function(s) {
    let result = 0;
    let set = new Set();

    if(s.length <= 1){
        return s.length;
    }

    for(let i = 0; i < s.length; i++){
        if(set.has(s[i])){
            result += 2;
            set.delete(s[i]);
        } else {
            set.add(s[i]);
        }
    }

    if(set.size >= 1){
        result++;
    }

    return result;
};

File: 409-longest-palindrome/longest-palindrome.c
int longestPalindrome(char* s) {
    int *arr;
    int result = 0;
    int remaining = 0;

    arr = (int*) calloc(56, sizeof(int));

    int index = 0;
    
    while(*(s + sizeof(char) * index) != '\0'){
        //printf("%c", *(s + sizeof(char) * index));
        //printf("%d", *(s + sizeof(char) * index) - 'A');
        int value = (int)(*(s + sizeof(char) * index) - 'A');
        if(value > 25){
            value -= 6;
        }
        if(*(arr + value) == 1){
            *(arr + value) = 0;
            result += 2;
            remaining--;
        } else {
            *(arr + value) = 1;
            remaining++;
        }
        index++;
    }

    if(remaining >= 1){
        result++;
    }

    return result;
}